images/devtools-terraform-v1beta1/context/Dockerfile

```diff
@@ -41,34 +43,36 @@ RUN \
 VOLUME /srv/workspace
 WORKDIR /srv/workspace
 
-ARG tfswitcher_version=0.13.1218
-
 COPY --from=terraform /bin/terraform /usr/bin/terraform
 COPY --from=tflint /root/.tflint.d /root/.tflint.d
 COPY --from=tflint /usr/local/bin/tflint /usr/bin/tflint
 COPY --from=tfsec  /usr/bin/tfsec /usr/bin/tfsec
 COPY --from=golang /usr/local/bin/goenv.sh /usr/local/bin/goenv.sh
 COPY --from=docker-lock /prod/docker-lock /usr/local/bin/docker-lock
 
-COPY terraform-switcher_${tfswitcher_version}_checksums.txt /var/tmp/terraform-switcher_${tfswitcher_version}_checksums.txt
-
 # Setting TF_PLUGIN_CACHE_DIR makes things a lot faster but this somehow
 # causes weird permission issues.
 # ENV TF_PLUGIN_CACHE_DIR=/root/.cache/terraform.d/plugin-cache
 
-ARG terraform_versions="0.15.5 1.0.11 1.1.7"
-
 RUN \
     mkdir -vp ~/.cache ~/.local/bin ~/bin && \
     true
 
-RUN \
-    cd /var/tmp && \
-    . /usr/local/bin/goenv.sh && \
-    curl -LOJ "https://github.com/warrensbox/terraform-switcher/releases/download/${tfswitcher_version}/terraform-switcher_${tfswitcher_version}_${GOOS}_${GOARCH}.tar.gz" && \
-    sed -i "/_${GOOS}_${GOARCH}.tar.gz/!d" "terraform-switcher_${tfswitcher_version}_checksums.txt" && \
-    sha256sum -c "terraform-switcher_${tfswitcher_version}_checksums.txt" && \
-    tar -xvf "terraform-switcher_${tfswitcher_version}_${GOOS}_${GOARCH}.tar.gz" -C /usr/local/bin tfswitch && \
+
+
+RUN --mount=type=secret,id=github_token \
+    GITHUB_OAUTH_TOKEN=$(cat /run/secrets/github_token) && \
+    ASSETS=$(curl -H "Accept: application/vnd.github.v3+json" -u "${GITHUB_OAUTH_TOKEN}:x-oauth-basic" "https://api.github.com/repos/warrensbox/terraform-switcher/releases/latest" | jq --raw-output ".assets[]") && \
+    TERRAFORM_SWITCHER_FILE_NAME=$(echo "${ASSETS}" | jq --raw-output "select(.name | contains(\"linux_amd64\")) | .name") && \
+    TERRAFORM_SWITCHER_URL=$(echo "${ASSETS}" | jq --raw-output "select(.name | contains(\"linux_amd64\")) | .browser_download_url") && \
+    TERRAFORM_SWITCHER_CHECKSUMS_FILE_NAME=$(echo "${ASSETS}" | jq --raw-output "select(.name | contains(\"checksums\")) | .name") && \
+    TERRAFORM_SWITCHER_CHECKSUMS_URL=$(echo "${ASSETS}" | jq --raw-output "select(.name | contains(\"checksums\")) | .browser_download_url") && \
+    curl -LO "${TERRAFORM_SWITCHER_URL}" && \
+    curl -LO "${TERRAFORM_SWITCHER_CHECKSUMS_URL}" && \
+    sed -i "/${TERRAFORM_SWITCHER_FILE_NAME}/!d" "${TERRAFORM_SWITCHER_CHECKSUMS_FILE_NAME}" && \
+    sha256sum -c "${TERRAFORM_SWITCHER_CHECKSUMS_FILE_NAME}" && \
+    tar -xvf "${TERRAFORM_SWITCHER_FILE_NAME}" -C /usr/local/bin tfswitch && \
+    rm -rf "${TERRAFORM_SWITCHER_FILE_NAME}" "${TERRAFORM_SWITCHER_CHECKSUMS_FILE_NAME}" && \
```

Comment:
This eliminates any meaningful digest verification as we are downloading the digest from the same place as the the checksums.

Reply:
I am just curious about where you downloaded digest before?

Reply:
From GitHub, but I did not do it at the same time as building the image, using the same machine that builds the image, and using the same network connection, which are all factors which reduces the attack surface area.

Reply:
> From GitHub, but I did not do it at the same time as building the image, using the same machine that builds the image, and using the same network connection, which are all factors which reduces the attack surface area.

If I will make the download of assets of a release via GH CLI will it be better?

Reply:
I don't really care about the technical detail as much as I care about us following our [guidelines](https://playbook.internal.coop/guidelines/security/supply_chain.html?h=lock#version-locking):

> Lock the versions of your dependencies. When locking the version of a dependency a checksum for the exact version of the dependency is store in your source code repository and used to reinstall the exact same version. This ensures that an upstream attack overwriting a valid version of the dependency is not installed by accident.

Reply:
Or alternatively, updating the guidelines if we don't feel they are useful/helpful.